src/Analyzer.Utilities.UnitTests/JsonLineNumberResolverTests.cs

```diff
@@ -342,5 +342,193 @@ public void Constructor_NullTemplateContext_ThrowsException()
+                                      }
+                                    }
+                                  ]
+                                }
+                              }
+                            }
+                          ]
+                        }"),
+                ExpandedTemplate = JObject.Parse(
+                    @"{
+                          ""$schema"": ""https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#"",
+                          ""contentVersion"": ""1.0.0.0"",
+                          ""resources"": [
+                            {
+                              ""type"": ""Microsoft.Resources/deployments"",
+                              ""apiVersion"": ""2019-10-01"",
+                              ""name"": ""nestedTemplate"",
+                              ""resourceGroup"": ""my-rg"",
+                              ""properties"": {
+                                ""mode"": ""Incremental"",
+                                ""expressionEvaluationOptions"": {
+                                  ""scope"": ""inner""
+                                },
+                                ""template"": {
+                                  ""$schema"": ""https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#"",
+                                  ""contentVersion"": ""1.0.0.0"",
+                                  ""parameters"": {
+                                    ""ftpsState"": {
+                                      ""type"": ""string"",
+                                      ""defaultValue"": ""undesiredValue""
+                                    }
+                                  },
+                                  ""resources"": [
+                                    {
+                                      ""apiVersion"": ""2019-08-01"",
+                                      ""type"": ""Microsoft.Web/sites"",
+                                      ""kind"": ""api"",
+                                      ""name"": ""withoutSpecifyingProperties"",
+                                      ""location"": ""US"",
+                                      ""properties"": {
+                                        ""siteConfig"": {
+                                          ""ftpsState"": ""[parameters('ftpsState')]""
+                                        }
+                                      }
+                                    }
+                                  ]
+                                }
+                              }
+                            }
+                          ]
+                        }"),
+                ResourceMappings = new Dictionary<string, string>
+                {
+                    { "resources[0]", "resources[0]" },
+                }
+            };
+            TemplateContext childTemplateContext = new TemplateContext
+            {
+                OriginalTemplate = JObject.Parse(
```

Comment:
Instead of duplicating this text, define it once in the test and extract the nested template to assign here.  Otherwise it's too easy to make a mistake in the test and have the text between the two become out of sync, which can result in a test passing when it shouldn't (meaning a bug gets introduced).